src/client/Client.cc

```diff
@@ -7971,10 +7971,10 @@ int Client::_setattr(InodeRef &in, struct stat *attr, int mask,
   stat_to_statx(attr, &stx);
   mask &= ~CEPH_SETATTR_BTIME;
 
-  if ((mask & CEPH_SETATTR_UID) && attr->st_uid == static_cast<uid_t>(-1)) {
+  if ((mask & CEPH_SETATTR_UID) && attr->st_uid == (short)static_cast<uid_t>(-1)) {
     mask &= ~CEPH_SETATTR_UID;
   }
-  if ((mask & CEPH_SETATTR_GID) && attr->st_gid == static_cast<uid_t>(-1)) {
+  if ((mask & CEPH_SETATTR_GID) && attr->st_gid == (short)static_cast<uid_t>(-1)) {
```

Comment:
How does attr->st_uid end up as a short here? src/include/compat.h has this when WIN32 is defined:
```
typedef unsigned int uid_t;
typedef unsigned int gid_t;
```
...and we definitely want more than 16 bits for uids/gids.

Also, the second condition should probably be using `static_cast<gid_t>`. Can you fix that up while you're in here?